arangod/RocksDBEngine/RocksDBVectorIndexList.cpp

```diff
@@ -159,7 +199,131 @@ bool RocksDBInvertedListsFilteringIterator::searchFilteredIds() {
   return true;
 }
 
-void RocksDBInvertedListsFilteringIterator::skipOverFilteredDocuments() {
+template<VectorIndexStoredValuesStrategy Strategy>
+void RocksDBInvertedListsFilteringIterator<
+    Strategy>::skipOverFilteredDocuments() {
+  while (_filteredIdsIt == _filteredIds.end()) {
+    if (!searchFilteredIds()) {
+      // If we enter here we could not produce any documents
+      return;
+    }
+  }
+}
+
+template<VectorIndexStoredValuesStrategy Strategy>
+void RocksDBInvertedListsFilteringIterator<Strategy>::next() {
+  skipOverFilteredDocuments();
+  ++_filteredIdsIt;
+  if (_filteredIdsIt == _filteredIds.end()) {
+    skipOverFilteredDocuments();
+  }
```

Comment:
We don't really need this `if` since it is implicit in the while-loop in skipOverFilteredDocuments.